apps/web/src/pages/quick-start/steps/Setup.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import styled from '@emotion/styled';
+import { useMutation } from '@tanstack/react-query';
+import { Stack, Stepper } from '@mantine/core';
+
+import { ChannelCTATypeEnum, ICreateNotificationTemplateDto, INotificationTemplate, StepTypeEnum } from '@novu/shared';
+
+import { QuickStartWrapper } from '../components/QuickStartWrapper';
+import { useNotificationGroup } from '../../../api/hooks/useNotificationGroup';
+import { useTemplates } from '../../../api/hooks/useTemplates';
+import { useEnvController } from '../../../store/useEnvController';
+import { useInAppActivated } from '../components/useInAppActivated';
+import { APPLICATION_IDENTIFIER, frameworkInstructions, notificationTemplateName } from '../consts';
+import { createTemplate } from '../../../api/notification-templates';
+import { LoaderProceedTernary } from '../components/LoaderProceedTernary';
+import { Prism } from '../../settings/tabs/components/Prism';
+import { When } from '../../../components/utils/When';
+import { colors } from '../../../design-system';
+
+export function Setup() {
+  const [notificationTemplate, setNotificationTemplate] = useState<INotificationTemplate>();
+  const { framework } = useParams();
+  const { groups } = useNotificationGroup();
+  const { templates = [], loading } = useTemplates();
+  const { environment } = useEnvController();
+  const { initialized } = useInAppActivated();
+
+  const instructions = frameworkInstructions.find((instruction) => instruction.key === framework)?.value ?? [];
+  const openBrowser = framework === 'demo';
+
+  const environmentIdentifier = environment?.identifier ? environment.identifier : '';
+
+  const { mutateAsync: createNotificationTemplate } = useMutation<
+    INotificationTemplate,
+    { error: string; message: string; statusCode: number },
+    ICreateNotificationTemplateDto
+  >(createTemplate);
+
+  useEffect(() => {
+    if (!loading) {
+      const onboardingNotificationTemplate = templates.find((template) =>
+        template.name.includes(notificationTemplateName)
+      );
+
+      if (onboardingNotificationTemplate) {
+        setNotificationTemplate(onboardingNotificationTemplate);
+      } else {
+        createOnBoardingTemplate();
+      }
+    }
+  }, [templates]);
+
+  async function createOnBoardingTemplate() {
+    const payloadToCreate = {
+      notificationGroupId: groups[0]._id,
+      name: notificationTemplateName,
+      active: true,
+      draft: false,
+      steps: [
+        {
+          template: {
+            type: StepTypeEnum.IN_APP,
+            content:
+              'Welcome to Novu! Click on this notification to <b>visit the cloud admin panel</b> managing this message',
+            cta: {
+              type: ChannelCTATypeEnum.REDIRECT,
+              data: { url: `/templates/edit/${notificationTemplate?._id}` },
```

Comment:
Need to think about the copy of the notification that will be sent to the user's bell

Reply:
Can we redirect the user back to the screen of the onboarding once clicked?

Reply:
I think we can but only if we add to all of the clients (and by clients I mean react/angular/vue/demo and the embed code snippets) onNotificationClick handler with window.location.href update.

Reply:
@scopsy let me know what you think about it.

Reply:
@scopsy i will merge the branch, but let me know what you think, maybe we can remove it.